src/wal/src/local_storage_impl/segment.rs

```diff
@@ -615,6 +615,32 @@ impl SegmentManager {
     }
 }
 
+pub struct SegmentTableMeta {
```

Comment:
I think we should define this in `tools`, not here.

Reply:
To minimize exposing the internal objects, I chose not to return all the segments directly. Furthermore, since all_segments contain Mutexs, exposing it through the interface could compromise both safety and maintainability.

Reply:
> To minimize exposing the internal objects, I chose not to return all the segments directly.

Sounds reasonable for me. I suggest we rename this struct to `SegmentView` to represent it's a static version of segment.

Reply:
Done.